containers/vuejs/src/components/PongCanvas.vue

```diff
@@ -6,26 +6,49 @@
 </template>
 
 <script setup lang="ts">
-import { ref, onMounted } from 'vue'
+import { onMounted, ref, watch } from 'vue'
 import { getSocketIOInstance } from './SocketManager'
 import GameHeader from './GameHeader.vue'
 
-const socketIOGame = getSocketIOInstance('game')
+import { useRoute } from 'vue-router'
+const route = useRoute()
+
+let socketIOGame: SocketIOClient.Socket | undefined
+
+// TODO: Move this to a logical file, and figure out a way to do this without watch()
+// TODO: Comment this back in!
+watch(
+  () => route.query.code,
+  (code) => {
+    if (code && socketIOGame === undefined) {
+      // TODO: Only call this when we return from a Login button, and if we have a query code
+      // TODO: This is an extremely shitty solution, since there's a race condition caused by
+      // TODO: PlayButton.vue also calling this function, so PLEASE think of something better :((
+      // TODO: Can the socketIOGame not just be a global?
```

Comment:
The socket is global, singleton pattern translates to global state. I don't want to expose it where it is not needed